lib/events/s3_importer.rb

```diff
@@ -0,0 +1,53 @@
+module Events
+  class S3Importer
+    def initialize(s3_key)
+      @s3_key = s3_key
+    end
+
+    def import
+      file = Zlib::GzipReader.new(object.get.body)
+      csv = CSV.new(file, headers: true)
```

Comment:
If you do CSV.parse(file, headers: true).each {|csv| stuff} you probably won't need to rewind?  
so:
```ruby
      csv = CSV.parse(file, headers: true).each do |row|
        event = Event.find_or_initialize_by(id: row["id"])
        event.update!(attributes(row))
      end
      csv.count
```